sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -49,4 +82,380 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
+            bypassCreatedBy: auditUser,
+            creator: auditUser,
+            createdAt: now,
+            bypassBehavior: payload.bypassBehavior,
+            creationNote: noteObj,
+          });
+      },
+    );
+  }
+
+  /**
+   * Checks if the application is in a valid state for bypass creation.
+   * Throws an error if the application is in a Draft, Cancelled,
+   * or Overwritten state.
+   * @param transactionalEntityManager transactional entity manager to execute the query.
+   * @param applicationId id of the application to check.
+   */
+  private async checkForApplicationInValidState(
+    transactionalEntityManager: EntityManager,
+    applicationId: number,
+  ) {
+    const application = await transactionalEntityManager
+      .getRepository(Application)
+      .findOne({
+        select: {
+          applicationStatus: true,
+        },
+        where: {
+          id: applicationId,
+        },
+      });
+    if (
+      [
+        ApplicationStatus.Draft,
+        ApplicationStatus.Cancelled,
+        ApplicationStatus.Overwritten,
+      ].includes(application.applicationStatus)
+    ) {
+      throw new CustomNamedError(
+        "Cannot create a bypass when application is in invalid state.",
+        APPLICATION_IN_INVALID_STATE_FOR_APPLICATION_RESTRICTION_BYPASS_CREATION,
+      );
+    }
+  }
+
+  /**
+   * Checks if a student restriction is active.
+   * Throws an error if the student restriction is not found
+   * or if it is not active.
+   * @param transactionalEntityManager transactional entity manager to execute the query.
+   * @param studentRestrictionId id of the student restriction to check.
+   */
+  private async checkForActiveStudentRestriction(
+    transactionalEntityManager: EntityManager,
+    studentRestrictionId: number,
+  ) {
+    const studentRestriction = await transactionalEntityManager
+      .getRepository(StudentRestriction)
+      .findOne({
+        select: {
+          isActive: true,
+        },
+        where: {
+          id: studentRestrictionId,
+        },
+      });
+    if (!studentRestriction) {
+      throw new CustomNamedError(
+        "Could not find student restriction for the given id.",
+        STUDENT_RESTRICTION_NOT_FOUND,
+      );
+    }
+    if (studentRestriction.isActive === false) {
+      throw new CustomNamedError(
+        "Cannot create a bypass when student restriction is not active.",
+        STUDENT_RESTRICTION_IS_NOT_ACTIVE,
+      );
+    }
+  }
+
+  /**
+   * Checks if there is an active application student restriction bypass.
+   * Throws an error if there is an active bypass for the same active student restriction id.
+   * @param transactionalEntityManager transactional entity manager to execute the query.
+   * @param applicationId id of the application to check.
+   * @param studentRestrictionId id of the student restriction to check.
+   */
+  private async checkForActiveApplicationStudentRestrictionByPasses(
```

Comment:
Bypass is a single word hence it should be `checkForActiveApplicationStudentRestrictionBypasses`, right?